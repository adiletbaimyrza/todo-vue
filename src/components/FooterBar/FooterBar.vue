<template>
  <footer :class="$style['container']">drag and drop to reorder list</footer>
</template>

<script>
export default {
  name: 'FooterBar',
}
</script>

<style lang="scss" module>
.container {
  margin-top: 4em;
  color: #b1b1b1;

  &::first-letter {
    text-transform: capitalize;
  }
}
</style>
