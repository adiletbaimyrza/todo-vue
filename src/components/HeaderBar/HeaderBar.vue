<template>
  <header :class="$style['bar']">
    <h1 :class="$style['bar__heading']">todo</h1>
    <button type="button" :class="$style['bar__button']">
      <CoreIcon
        :src="require('../../assets/moon.svg')"
        height="32px"
        width="32px"
      />
    </button>
  </header>
</template>

<script>
import CoreIcon from '../CoreIcon/CoreIcon.vue'

export default {
  name: 'HeaderBar',
  components: {
    CoreIcon,
  },
}
</script>

<style lang="scss" module>
.bar {
  display: flex;
  justify-content: space-between;
  width: 600px;
  margin-bottom: 4em;
  margin-top: 16vh;

  &__heading {
    text-transform: uppercase;
    color: #ffffff;
    letter-spacing: 0.5em;
    cursor: default;
  }
}
</style>
