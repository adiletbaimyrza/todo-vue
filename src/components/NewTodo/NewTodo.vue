<template>
  <div :class="$style['bar']">
    <CoreIcon
      :src="require('../../assets/circle.svg')"
      :class="$style['bar__icon']"
    />
    <input
      v-model="newTodoContent"
      @keyup.enter="addNewTodo"
      :class="$style['bar__input']"
      type="text"
      placeholder="Ex: Buy groceries..."
    />
  </div>
</template>

<script>
import CoreIcon from '../CoreIcon/CoreIcon.vue'

export default {
  name: 'NewTodo',
  components: { CoreIcon },
  data() {
    return {
      newTodoContent: '',
    }
  },
  methods: {
    addNewTodo() {
      if (this.newTodoContent.trim() !== '') {
        const newTodo = {
          id: Date.now(),
          content: this.newTodoContent,
          isCompleted: false,
        }
        this.$store.commit('add', newTodo)
        this.newTodoContent = ''
      }
    },
  },
}
</script>

<style lang="scss" module>
.bar {
  display: flex;
  width: 600px;
  background-color: var(--card-background-color);
  padding: 1em 1.5em;
  border-radius: 0.4em;
  margin-bottom: 1.5em;
  box-shadow: rgba(100, 100, 111, 0.2) 0px 7px 29px 0px;
  caret-color: var(--font-color-regular);

  &__icon {
    margin-right: 1em;
  }

  &__input {
    flex-grow: 1;
    outline: none;
    border: none;
    font-size: 1em;
    color: var(--font-color-regular);
    background-color: var(--card-background-color);

    &::placeholder {
      color: #b1b1b1;
    }
  }
}
</style>
