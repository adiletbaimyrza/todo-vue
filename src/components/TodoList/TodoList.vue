<template>
  <div :class="$style['list']">
    <TodoItem
      v-for="item in filteredTodos"
      :key="item.id"
      :id="item.id"
      :content="item.content"
      :isCompleted="item.isCompleted"
    />
    <FilterBar />
  </div>
</template>

<script>
import { mapGetters } from 'vuex'
import FilterBar from '../FilterBar/FilterBar.vue'
import TodoItem from '../TodoItem/TodoItem.vue'

export default {
  name: 'TodoList',
  components: {
    TodoItem,
    FilterBar,
  },
  computed: {
    ...mapGetters(['filteredTodos']),
  },
}
</script>

<style lang="scss" module>
.list {
  display: flex;
  flex-direction: column;
  box-shadow: rgb(100 100 111 / 20%) 0 7px 29px 0;
  border-radius: 0.4em;

  > :first-child {
    border-top-left-radius: 0.4em;
    border-top-right-radius: 0.4em;
  }

  > :last-child {
    border-bottom-left-radius: 0.4em;
    border-bottom-right-radius: 0.4em;
    border-bottom: none;
  }
}
</style>
