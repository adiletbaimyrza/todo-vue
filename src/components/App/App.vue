<template>
  <HeaderBar />
  <NewTodo />
  <TodoList />
  <FooterBar />
</template>

<script>
import HeaderBar from '../HeaderBar/HeaderBar.vue'
import NewTodo from '../NewTodo/NewTodo.vue'
import TodoList from '../TodoList/TodoList.vue'
import FooterBar from '../FooterBar/FooterBar.vue'

export default {
  name: 'App',
  components: { HeaderBar, NewTodo, TodoList, FooterBar },
}
</script>

<style lang="scss">
:root {
  --app-background-color: #ffffff;
  --card-background-color: #ffffff;
  --font-color-regular: #000000;
  --filter-text--hover: #000000;
  --card-background-color--hover: #f1f1f1;
  --card-border-color: #e4e4e4;
}

[theme='dark'] {
  --app-background-color: #091a28;
  --card-background-color: #091a28;
  --font-color-regular: #ffffff;
  --filter-text--hover: #ffffff;
  --card-background-color--hover: #11314b;
  --card-border-color: #08263f;
}

/* css reset --- start */
*,
*::before,
*::after {
  margin: unset;
  padding: unset;
  box-sizing: border-box;
  z-index: 3;
}

button {
  border: unset;
  background: unset;
  cursor: pointer;
}

ul,
ol {
  list-style: none;
}
/* css reset --- end */

@import url('https://fonts.googleapis.com/css2?family=Inter:ital,opsz,wght@0,14..32,100..900;1,14..32,100..900&display=swap');

html,
body,
#app {
  width: 100vw;
  height: 100vh;
  transition: color 0.1s linear, background 0.1s linear,
    background-color 0.1s linear;
}

#app {
  display: flex;
  flex-direction: column;
  align-items: center;
  font-family: Inter, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  text-align: center;
  overflow-x: hidden;
  background-color: var(--app-background-color);
  z-index: 0;

  &::before {
    content: '';
    position: absolute;
    top: 0;
    width: 100%;
    height: 45%;
    background-size: cover;
    background-image: linear-gradient(
        142deg,
        rgba(189, 58, 208, 0.7198529069831057) 30%,
        rgba(153, 90, 194, 0.5237744756105567) 55%,
        rgba(38, 121, 180, 0.6526260162268032) 79%
      ),
      url('../../assets/mountains.jpg');
    background-position: center;
    background-repeat: no-repeat;
    z-index: 1;
  }
}
</style>
