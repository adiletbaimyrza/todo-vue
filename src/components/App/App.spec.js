import { shallowMount } from '@vue/test-utils'
import App from './App.vue'
import HeaderBar from '../HeaderBar/HeaderBar.vue'
import NewTodo from '../NewTodo/NewTodo.vue'
import TodoList from '../TodoList/TodoList.vue'
import FooterBar from '../FooterBar/FooterBar.vue'

/* global describe it expect beforeEach */
describe('App.vue', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(App)
  })

  it('renders the App correctly', () => {
    expect(wrapper.exists()).toBe(true)
    expect(wrapper.element).toMatchSnapshot()
  })

  it('renders the HeaderBar component', () => {
    const headerBar = wrapper.findComponent(HeaderBar)
    expect(headerBar.exists()).toBe(true)
  })

  it('renders the NewTodo component', () => {
    const newTodo = wrapper.findComponent(NewTodo)
    expect(newTodo.exists()).toBe(true)
  })

  it('renders the TodoList component', () => {
    const todoList = wrapper.findComponent(TodoList)
    expect(todoList.exists()).toBe(true)
  })

  it('renders the FooterBar component', () => {
    const footerBar = wrapper.findComponent(FooterBar)
    expect(footerBar.exists()).toBe(true)
  })
})
