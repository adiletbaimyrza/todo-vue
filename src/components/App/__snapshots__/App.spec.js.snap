// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue renders the App correctly 1`] = `
<div
  data-v-app=""
>
  
  <header-bar-stub />
  <new-todo-stub />
  <todo-list-stub />
  <footer-bar-stub />
  
</div>
`;
